test-data/unit/check-multiple-inheritance.test

```diff
@@ -240,3 +240,37 @@ def dec(f: Callable[..., T]) -> Callable[..., T]:
 [out]
 main:3: error: Cannot determine type of 'f' in base class 'B'
 main:3: error: Cannot determine type of 'f' in base class 'C'
+
+[case testMultipleInheritanceNestedClassesWithSameNameAllowed]
+class Mixin1:
+    class Meta:
+        pass
+class Mixin2:
+    class Meta:
+        pass
+class A(Mixin1, Mixin2):
+    pass
+[out]
+
+[case testMultipleInheritanceFailIfNestedClassAndAttrHaveSameName]
+class Mixin1:
+    class Nested1:
+        pass
+class Mixin2:
+    Nested1: str
+class A(Mixin1, Mixin2):
+    pass
+[out]
+main:6: error: Definition of "Nested1" in base class "Mixin1" is incompatible with definition in base class "Mixin2"
+
+[case testMultipleInheritanceFailIfNestedClassAndFunctionHaveSameName]
+class Mixin1:
+    class Nested1:
+        pass
+class Mixin2:
+    def Nested1(self) -> str:
+        pass
+class A(Mixin1, Mixin2):
+    pass
+[out]
+main:7: error: Definition of "Nested1" in base class "Mixin1" is incompatible with definition in base class "Mixin2"
```

Comment:
Please add test cases where one of the classes defines a variable referring to another class, for example:
```python
class A:
    def __init__(self, arg: str) -> None: ...
class B(A): ...

class Base1:
    Nested = A
class Base2:
    Nested = B
class Combo(Base1, Base1): ...
```
I think this test case (and similar) should pass. Since this is logically (almost) equivalent to "inline" class definitions.